src/middlewared/middlewared/plugins/zettarepl.py

```diff
@@ -213,9 +218,13 @@ def _observer(self, message):
 
                 if isinstance(message, (PeriodicSnapshotTaskSuccess, PeriodicSnapshotTaskError)):
                     context = self.vmware_contexts.pop(task_id, None)
-                    if context:
+                    vm_context = self.vm_contexts.pop(task_id, None)
+                    if context or vm_context:
                         with Client() as c:
-                            c.call("vmware.periodic_snapshot_task_end", context, job=True)
+                            if context:
+                                c.call("vmware.periodic_snapshot_task_end", context, job=True)
+                            if vm_context:
+                                c.call("vm.resume_suspended_vms", list(vm_context))
```

Comment:
Are we sure this call will never exceed the timeout of 60 seconds?

Reply:
The operation is quite quick, I would say if we have around 20-30 vms being suspended, it should be fine. Should we perhaps change these methods to a job ?